governance/program/src/processor/process_deposit_governing_tokens.rs

```diff
@@ -52,13 +57,32 @@ pub fn process_deposit_governing_tokens(
         governing_token_holding_info.key,
     )?;
 
-    transfer_spl_tokens(
-        governing_token_source_info,
-        governing_token_holding_info,
-        governing_token_transfer_authority_info,
-        amount,
-        spl_token_info,
-    )?;
+    let realm_config_data =
+        get_realm_config_data_for_realm(program_id, realm_config_info, realm_info.key)?;
+
+    realm_config_data.assert_can_deposit_governing_token(&realm_data, &governing_token_mint)?;
```

Comment:
Setting token type as `Dormant` has no direct security implications. It only allows to make the intention of not using token deposits as governance power stronger and explicit. This could prevent attacks in case of unintentional code changes in the future.

The `Dormant` token type is used in two use cases: 

1. When only a single governance population is used. For example a `Multisig` DAO can utilise only the `Council` token for governance power and  the `Community` token is not used yet. Once the DAO decentralises then it would grant governance power to the `Community` but in the meantime it keeps it dormant.

     It's possible a DAO would like to stay `Multisig` forever with no intention of decentralisation. In this case the stronger guaranteed would be to freeze the `Community` token with 0 `supply` but I prefer to keep the options open for features like social recovery and hence setting the token as `Dormant` feels like a better option.

1.   Plugins which take tokens deposits. The plugins like `VSR` (Mango `veToken` locker) take tokens deposits and hence the tokens deposited into the `Realm` are not used. In the original design when a plugin was configured then it would have also disabled deposits. However plugins which depend on `Realm` deposits have been developed (ex. `Civic Gateway`) and we had to loosen the constraint https://github.com/solana-labs/solana-program-library/pull/3311

  In cases like the `VSR` plugin setting the token type as `Dormant` makes the intention explicit, but at the same time it's a reversible action and the tokens deposits into `Realm` could be used again if the DAO choses so.

I updated comments in the code to explain the use case better.